artifacts-download-trigger/bin/downloader/action.yml

```diff
@@ -0,0 +1,57 @@
+name: artifact downloader
+description: downloader and extract an artifact associated with given workflow and commit
+author: Josef Fröhle <github@josef-froehle.de>
+branding:
+  icon: download
+  color: green
+inputs:
+  gh-token:
+    description: 'GitHub token with repo write access, can NOT use secrets.GITHUB_TOKEN, see readme'
+    required: true
+  workflow:
+    description: 'Name or ID of workflow to run'
+    required: true
+  workflow-conclusion:
+    description: |
+      Wanted status or conclusion to search for in recent runs
+      https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-workflow-runs
+    required: false
+    default: success
+  repo:
+    description: Repository name with owner (like actions/checkout)
+    required: false
+    default: ${{github.repository}}
+  pr:
+    description: Pull request number
+    required: false
+  commit:
+    description: Commit hash
+    required: false
+  branch:
+    description: Branch name
+    required: false
+  event:
+    description: Event type
+    required: false
+  run-id:
+    description: Workflow run id
+    required: false
+  run-number:
+    description: Workflow run number
+    required: false
```

Comment:
**pr** think about automatically test, and the result on a PR, that can downloaded with it.
**commit** if you have a specific commit that you need for older versions, think about you have changed the structure of the results it can rerun the commit and the workflows on it until the product is EOL. and you stop the runs.
**branch** i don't think I need to talk about it....
**event** if you have the same commit/branch etc. but you have a workflow where you need something like manual triggeres results, then you use it and not the one that is scheduled or a pr or push...
**run-id** use only the results of this run with the run-id, together with run-number  that count up if you rerun the workflow run with the same run-id. Typical if your use external data that can change, you can rerun the same workflow run ever and ever again... 

We have currently nothing expect of the branch in use, because we currently have only simple workflows.